### PR TITLE
test refactor

### PR DESCRIPTION
**Pull request by rabernat**
I refactored @sambarluc's test into one single test, which mysteriously is now passing on my machine. I want to see the result on travis.

### CODE REVIEW

**Comment by rabernat** [comment]
@sambarluc: here the test is failing only on python 2 and only with swap_dims=True.
https://travis-ci.org/xgcm/xmitgcm/jobs/289577976

It is passing for python 3 (now that we don't try to swap dims on llc grids), as I suggested in my final comment in #53. 

Do you know how to check out this PR? Can you see if the test passes on your machine?

**Comment by sambarluc** [comment]
Not really sure on how to checkout the pr, I'm really getting lost with the complexity of git.

In any case, I suspect the test may now be insufficient, probably we should try printing the time axis?

**Comment by sambarluc** [comment]
I actually added the tests with the print, but I am not even sure of where they are. It takes devotion to learn git.

File: xmitgcm/test/test_mds_store.py [hunk 1]
```diff
@@ -749,44 +749,22 @@ def test_drc_length(all_mds_datadirs):
...
-    dirname, expected = mds_datadirs_with_refdate
-
-    ds = xmitgcm.open_mdsdataset(dirname, iters='all', prefix=['S'],
-                                 ref_date=expected['ref_date'], read_grid=True,
-                                 delta_t=expected['delta_t'], swap_dims=method,
-                                 geometry=expected['geometry'])
+                                 geometry=expected['geometry'],
+                                 read_grid=read_grid, swap_dims=swap_dims)
+    if load:
+        ds.time.load()
```
Comment (rabernat) [inline xmitgcm/test/test_mds_store.py]: @sambarluc: The time dimension gets loaded here. Is this not enough to trigger the error? It only happens with `print`, not `load`?

Comment (rabernat) [reply xmitgcm/test/test_mds_store.py]: Don't worry, your git is fine! I simplified your four tests functions into one single function with parametrize that does everything the original tests do (except try to call `swap_dims` with llc geometry).

**Comment by rabernat** [comment]
Don't worry, your git is fine! I simplified your four tests functions into one single function with parametrize that does everything the original tests do (except try to call `swap_dims` with llc geometry).

File: xmitgcm/test/test_mds_store.py (hunk 1 above)
Comment (sambarluc) [reply xmitgcm/test/test_mds_store.py]: Actually you are right, load is sufficient. But then I don't get how to get the version running on the server on my machine.

**Comment by rabernat** [comment]
https://help.github.com/articles/checking-out-pull-requests-locally/

**Comment by sambarluc** [comment]
Should have been more specific: I cannot seem to be able to find out what I should use as PR ID!

**Comment by rabernat** [comment]
```
git checkout -b rabernat-refactor_ref_date_tests master
git pull https://github.com/rabernat/xmitgcm.git refactor_ref_date_tests
```

**Comment by sambarluc** [comment]
Thanks.

It's definitely failing here:
```
xmitgcm/test/test_mds_store.py::test_tload_with_ref_date_grid[internal_wave-True] FAILED
xmitgcm/test/test_mds_store.py::test_tload_with_ref_date_grid[internal_wave-False] PASSED
xmitgcm/test/test_mds_store.py::test_tload_with_ref_date_swap[internal_wave-True] FAILED
xmitgcm/test/test_mds_store.py::test_tload_with_ref_date_swap[internal_wave-False] PASSED
xmitgcm/test/test_mds_store.py::test_open_with_ref_date_grid[internal_wave-True] FAILED
xmitgcm/test/test_mds_store.py::test_open_with_ref_date_grid[internal_wave-False] PASSED
xmitgcm/test/test_mds_store.py::test_open_with_ref_date_swap[internal_wave-True] FAILED
xmitgcm/test/test_mds_store.py::test_open_with_ref_date_swap[internal_wave-False] PASSED
xmitgcm/test/test_mds_store.py::test_date_parsing[global_oce_llc90] PASSED
xmitgcm/test/test_mds_store.py::test_tload_with_ref_date_grid[global_oce_llc90-True] PASSED
xmitgcm/test/test_mds_store.py::test_tload_with_ref_date_grid[global_oce_llc90-False] PASSED
xmitgcm/test/test_mds_store.py::test_tload_with_ref_date_swap[global_oce_llc90-True] FAILED
xmitgcm/test/test_mds_store.py::test_tload_with_ref_date_swap[global_oce_llc90-False] PASSED
xmitgcm/test/test_mds_store.py::test_open_with_ref_date_grid[global_oce_llc90-True] PASSED
xmitgcm/test/test_mds_store.py::test_open_with_ref_date_grid[global_oce_llc90-False] PASSED
xmitgcm/test/test_mds_store.py::test_open_with_ref_date_swap[global_oce_llc90-True] FAILED
xmitgcm/test/test_mds_store.py::test_open_with_ref_date_swap[global_oce_llc90-False] PASSED

```

**Comment by sambarluc** [comment]
I guess this means we are hitting some corner case related to small version differences?

**Comment by rabernat** [comment]
You have not actually checkout out this branch, because the test output doesn't look right. On my branch, I see the following:

```
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-True-True-True] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-True-True-False] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-False-True-True] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-False-True-False] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-False-False-True] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-False-False-False] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[global_oce_llc90-True-True-True] SKIPPED
xmitgcm/test/test_mds_store.py::test_ref_date[global_oce_llc90-True-True-False] SKIPPED
xmitgcm/test/test_mds_store.py::test_ref_date[global_oce_llc90-False-True-True] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[global_oce_llc90-False-True-False] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[global_oce_llc90-False-False-True] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[global_oce_llc90-False-False-False] PASSED
```

**Comment by sambarluc** [comment]
Strange, because I used the commands you sent me, and this looks right to me:

```
cimatori@ecolpc18:xmitgcm_sambarluc$ git status 
On branch rabernat-refactor_ref_date_tests
nothing to commit, working directory clean
```

I have no idea. I keep seeing things I don't understand on my fork. I suspect I messed up something at some point, but have no clue of what, nor on how to fix it. I'll try to see if I find a git master around.

**Comment by rabernat** [comment]
did you run `git pull https://github.com/rabernat/xmitgcm.git refactor_ref_date_tests`

**Comment by sambarluc** [comment]
Ok, I had to go far back in time and then come back, but here are the results. I checked that the test is indeed your refactored version:
```
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-True-True-True] FAILED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-True-True-False] FAILED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-False-True-True] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-False-True-False] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-False-False-True] PASSED
xmitgcm/test/test_mds_store.py::test_ref_date[internal_wave-False-False-False] PASSED

```

**Comment by sambarluc** [comment]
So, it fails with `swap_dims=True` only also here, even if it's python 3.6. Since I have an environment with 2.7 too, I checked and it fails in the same way there too.

**Comment by sambarluc** [comment]
Hi, I'm re-emerging from some stuff I had to do. Any news on this?

**Comment by rabernat** [comment]
I have not worked on it yet, and I doubt I will have much time soon. I think you are in a better position than me to fix this issue! Just dig in and you will probably find the solution!

**Comment by sambarluc** [comment]
I'll try, I just wanted to check, thanks!

**Comment by rabernat** [comment]
I'm going to merge this so we have the new tests in place.

**Comment by sambarluc** [comment]
Didn't have the time yet to really look into the issue, it's a bit obscure to me.

**Comment by sambarluc** [comment]
After another round of updates of my conda installation, the problem disappeared. Both in the test suite and in normal use. Embarrassingly enough, I did not track all the updates, but I can find out what was changed if you think it is necessary. Otherwise, I would say it was definitely some upstream issue, now fixed.

**Comment by sambarluc** [comment]
I would suggest to close the issue, but keep the tests for future reference.
